grammar.js

```diff
@@ -807,16 +833,13 @@ module.exports = grammar({
     )),
 
     enumerators: $ => seq(
-      sep1($._semicolon, choice(
-        $.enumerator,
-        $.assignment_expression
-      )),
+      sep1($._semicolon, $.enumerator),
       optional($._automatic_semicolon)
     ),
 
     enumerator: $ => seq(
-      $.identifier,
-      '<-',
+      $._pattern,
```

Comment:
while we're at it, should `$.assignment_expression` itself be using `$._pattern` for its LHS too?